features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpTelemetryAdapter.java

```diff
@@ -91,12 +118,51 @@ public BmpTelemetryAdapter(final AdapterDefinition adapterConfig,
         final Transport.StatisticsReportPacket stats = message.getStatisticsReport();
 
         // Find the node for the router who has exported the stats and build a collection agent for it
-        final InetAddress exporterAddress = InetAddressUtils.getInetAddress(messageLog.getSourceAddress());
-        final Optional<Integer> exporterNodeId = this.interfaceToNodeCache.getFirstNodeId(messageLog.getLocation(), exporterAddress);
+        InetAddress exporterAddress = InetAddressUtils.getInetAddress(messageLog.getSourceAddress());
+        Optional<Integer> exporterNodeId = this.interfaceToNodeCache.getFirstNodeId(messageLog.getLocation(), exporterAddress);
+
         if (!exporterNodeId.isPresent()) {
-            LOG.warn("Unable to find node and interface for agent address: {}", exporterAddress);
-            return Stream.empty();
+            LOG.warn("Unable to find node for exporter address: {}", exporterAddress);
+
+            if (message.hasBgpId()) {
+                final InetAddress bgpId = address(message.getBgpId());
+
+                final ExporterInfo exporterInfo = transactionTemplate.execute(new TransactionCallback<ExporterInfo>() {
+                    @Override
+                    public ExporterInfo doInTransaction(final TransactionStatus transactionStatus) {
+                        final List<OnmsNode> nodes = nodeDao.findNodeWithMetaData(contextKey.getContext(), contextKey.getKey(), InetAddressUtils.toIpAddrString(bgpId));
+
+                        if (nodes.size() > 0) {
```

Comment:
```suggestion
                        if (!nodes.isEmpty()) {
```

This one is for @mattixtech :)

Reply:
done.